docs-markdown/viz.md

```diff
@@ -0,0 +1,284 @@
+Update map template variables specific to heatmap visual
+
+[View options](https://github.com/mapbox/mapboxgl-jupyter/blob/add-markdown-docs/docs-markdown/viz.md#params-4)
+
+### Usage
+```python
+import pysal.esda.mapclassify as mapclassify
+import pandas as pd
+import os
+from mapboxgl.utils import *
+from mapboxgl.viz import *
+
+# create pandas df from csv file
+df = pd.read_csv('points.csv')
+
+# Must be a public token, starting with `pk`
+token = os.getenv('MAPBOX_ACCESS_TOKEN')
+
+# Create a geojson file export from a Pandas dataframe
+df_to_geojson(df, filename='points.geojson',
+              properties=['Avg Medicare Payments', 'Avg Covered Charges', 'date'],
+                     lat='lat', lon='lon', precision=3)
+
+#Create a heatmap
+heatmap_color_stops = create_color_stops([0.01,0.25,0.5,0.75,1], colors='RdPu')
+heatmap_radius_stops = [[0,1], [15, 40]] #increase radius with zoom
+
+color_breaks = mapclassify.Natural_Breaks(df['Avg Medicare Payments'], k=8, initial=0).bins
+color_stops = create_color_stops(color_breaks, colors='Spectral')
+
+heatmap_weight_stops = create_weight_stops(color_breaks)
+
+#Create a heatmap
+viz = HeatmapViz('points.geojson',
+                  access_token=token,
+                  weight_property = "Avg Medicare Payments",
+                  weight_stops = heatmap_weight_stops,
+                  color_stops = heatmap_color_stops,
+                  radius_stops = heatmap_radius_stops,
+                  opacity = 0.9,
+                  center = (-95, 40),
+                  zoom = 3,
+                  below_layer='waterway-label'
+                )
+
+viz.show()
+```
+![screen shot 2018-02-21 at 3 34 55 pm](https://user-images.githubusercontent.com/11286381/36511775-cfc4d794-171c-11e8-86b9-5f1a6060a387.png)
+
+
+## class MapViz
+
+The `MapViz` class is the base class of the various `mapboxgl-jupyter` visualizations. You can use this class to set default values for all visualizations rather than calling them directly from the other visualization objects.
+
+### Params
+**MapViz**(_data, access_token=None, center=(0, 0), below_layer='', opacity=1, div_id='map', height='500px', style_url='mapbox://styles/mapbox/light-v9?optimize=true', width='100%', zoom=0, min_zoom=0, max_zoom=24_)
+
+Parameter | Description
+--|--
+data | GeoJSON Feature Collection
+access_token | Mapbox GL JS access token.
+center | map center point
+style_url | url to mapbox style
+div_id | The HTML div id of the map container in the viz
+width | The CSS width of the HTML div id in % or pixels.
+height | The CSS height of the HTML map div in % or pixels.
+zoom | starting zoom level for map
+opacity | opacity of map data layer
+
+### Methods
+**as_iframe**(_self, html_data_)
+Build the HTML representation for the mapviz
+
+**show**(_self, **kwargs_)
+Load the HTML iframe
+
+**add_unique_template_variables**(_self, options_)
+
+**create_html**(_self_)
+Create a circle visual from a geojson data source
```

Comment:
Build the HTML text representation of the visual.  The output of this is a valid HTML document containing the visual object.